CCP-1822: Initial implementation

dqdquint opened the pull request:
Implemented a configurable video/image widget.

Set up a test widget in shop no: 31887.

Had a couple of different problems when implementing this feature.
- I tried used the media slider we have in engage but that proved to be unsuccessful since media is only gained if it is on the product that is passed.
- I tried using the MediaVideo component within the slider i created in the extension but this proved to be buggy in testing with looped videos not rendering on the first loop.
- I opted to go with using img and iframe tags to render slides and this works, except with auto play it only turns off when interacting with the next or prev buttons. I tried setting a ref to the iframe, but was unsuccessful in getting a click in the iframe to pause auto play in the slider. There is a method on the slider that we can call to halt auto play. 
- In testing this auto play issue didn't prove to be critical because when a user plays a video the video opens out of the slider in full screen.

Commit:
CCP-1822: Initial implementation

Commit:
CCP-1822: fixed casing issue

Carsten89 reviewed (changes requested):
(this is not a real review)
Please add:
- support for german lang
- /demo folder

Commit:
CCP-1822: Added demo folder, de translation, and autoplay config

aaron-martin commented:
Custom is misspelled in the description of the ConfigurableCarousel widget

aaron-martin commented:
the alignment on the buttons seem off to me. I think they should be vertically aligned middle within the gray area. This is only a problem in the Android app because it shows the gray area.
![image](https://user-images.githubusercontent.com/11764719/69074313-b9acc800-09f4-11ea-893c-fee7f8ad5a64.png)

aaron-martin commented:
On android the videos don't open in full screen so it is possible to slide away from the video while it is playing.

aaron-martin commented on frontend/components/VideoCarousel/index.jsx:
```diff
@@ -0,0 +1,157 @@
+  };
+
+  const slides = slideInfo.map((slide, index) => {
+    const item = slide.type === 'video' ?
```
I am not convinced that a ternary expression is the most readable choice here. I think this is more readable. 

```
  const slides = slideInfo.map((slide, index) => {
    if (slide.type === 'video') {
      return (
        <div
          className={styles.videoWrapper(backgroundColor)}
          key={index.toString()}
        >
          <div className={styles.videoResponsive}>
            <iframe className="swiper-lazy" title={slide.alt} data-src={slide.src}/>
            <div className="swiper-lazy-preloader swiper-lazy-preloader-black"/>
          </div>
        </div>
      );
    }
    return (
      <div
        key={index.toString()}
      >
        <img
          src={slide.src}
          alt={slide.alt}
        />
      </div>
    );
  });
```

dqdquint replied:
Updated thanks!

dqdquint commented:
@aaron-martin  I added additional margin to the carousel buttons, corrected the typo in ConfigurableCarousel , and disabled autoplay to address the video play issue after I confirmed that was reasonable with veronica.

Commit:
CCP-1822: addded margin for carousel buttons, corrected typo, and disabled autoplay

Commit:
fixed typo in description

aaron-martin commented on demo/index.md:
```diff
@@ -0,0 +1,85 @@
+- `numberSlidesPerView`: Expects a number. Sets the number of slides in view
+- `spaceBetweenSlides`: Expects a number. Sets the space between slides
+- `slideInfo`: Expects an array containing information for each slide in the carousel
+  - `alt`: Expects a string that provies a alternative titled for a slide
```
provides is misspelled

dqdquint replied:
Updated. Thanks!

aaron-martin commented on demo/index.md:
```diff
@@ -0,0 +1,85 @@
+    "spaceBetweenSlides": 30,
+    "slideInfo": [ 
+      { 
+        "alt":"monin 1",
```
I don't think we should use the Monin name here. Please replace this with something more generic.

dqdquint replied:
Updated

aaron-martin reviewed (changes requested):
just a few other small changes.

Commit:
CCP-1822: Updated demo file
